### PR TITLE
OCPBUGS-7488: test flake: should not reconcile SC when state is Unmanaged

### PR DESCRIPTION
**Pull request by dobsonj**
https://issues.redhat.com/browse/OCPBUGS-7488
/cc @openshift/storage

### CODE REVIEW

File: test/extended/storage/storageclass.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ var _ = g.Describe("[sig-storage][Feature:DisableStorageClass][Serial]", func()
 		sctest.SetAllowExpansion(false)
 
 		g.By("verifying the AllowVolumeExpansion reverts to true")
-		sctest.VerifyAllowExpansion(true)
+		sctest.VerifyAllowExpansion(true, func() {})
```
Comment (jsafrane) [inline test/extended/storage/storageclass.go:73]: this looks bad to me, can you pass `nil` and handle it inside `VerifyAllowExpansion`?

File: test/extended/storage/storageclass.go [hunk 2]
```diff
@@ -209,6 +212,8 @@ func (d *DisableStorageClassTest) VerifyAllowExpansion(expected bool) {
 		if allowExpansion == expected {
 			e2e.Logf("AllowVolumeExpansion matched %t after %d attempts", expected, maxRetries)
 			return
+		} else {
+			retry()
```
Comment (jsafrane) [inline test/extended/storage/storageclass.go:216]: Please log a retry + nr. of attempts.

File: test/extended/storage/storageclass.go (hunk 1 above)
Comment (dobsonj) [reply test/extended/storage/storageclass.go]: Fixed

**Comment by gnufied** [comment]
Yep, this could be hard to test. If it flakes we could delete `Unmanaged` tests entirely. IMO - a failing test is worse than having no test. :-)

**Pull request merged**